js/src/ipc/metadata.ts

```diff
@@ -53,7 +52,7 @@ export class RecordBatchMetadata extends Message {
     public buffers: BufferMetadata[];
     constructor(version: MetadataVersion, length: Long | number, nodes: FieldMetadata[], buffers: BufferMetadata[], bodyLength?: Long | number) {
         if (bodyLength === void(0)) {
-            bodyLength = buffers.reduce((s, b) => align(s + b.length + (b.offset - s), 8), 0);
+            bodyLength = buffers.reduce((bodyLength, buffer) => bodyLength + buffer.length, 0);
```

Comment:
Are your buffer lengths here guaranteed to a multiple of 8? Otherwise should round up

Reply:
Yep, this size is rounded up in the writer [here](https://github.com/apache/arrow/blob/2095e4ebffeb9f51f04d1b9500c958dbbca9bedd/js/src/ipc/writer/binary.ts#L298)